uv.lock

```diff
@@ -610,6 +668,22 @@ wheels = [
     { url = "https://files.pythonhosted.org/packages/45/07/95c6be6710207f4a2c83023f59889a86c58282c8367aa49a907e9e6deb57/plum_dispatch-2.2.2-py3-none-any.whl", hash = "sha256:d7ee415bd166ffa90eaa4b24d7c9dc7ca6f8875750586001e7c9baff706223bd", size = 30137 },
 ]
 
+[[package]]
+name = "pre-commit"
+version = "3.8.0"
+source = { registry = "https://pypi.org/simple" }
+dependencies = [
+    { name = "cfgv" },
```

Comment:
Why the heck does `pre-commit` have all these dependencies.. sigh.

Reply:
Pre-commit is powerful tool and is designed to handle much wider range of tools and use-cases. I added it as a dev dependency, so it shouldn’t be an issue.

Reply:
We currently install deps using `uv pip sync` because of some legacy reasons, so it installs everything :sweat_smile: But it doesn't really matter.

Reply:
In my Dockerfile I added parameter which excludes them. So after you switch , no more dev dependencies 😅